### PR TITLE
Allow duplicate orgs in different states; remove unused endpoints

### LINKED ISSUE #2388: Delete old/unused endpoints in AccountRequestController
### Description
Now that the signup form has been moved into the app and Experian is turned on, we can remove some superfluous endpoints in AccountRequestController.

### Proposed solution 

- [ ] Remove [`submitAccountRequest`](https://github.com/CDCgov/prime-simplereport/blob/61fed493759f7eaf445b3b737e32fafe3c66647c/backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java#L99)
- [ ] Remove [`submitAccountRequestV2`](https://github.com/CDCgov/prime-simplereport/blob/61fed493759f7eaf445b3b737e32fafe3c66647c/backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java#L165) 
- [ ] add a comment to `submitAccountRequestWithoutFacility` that it's not currently in use, but that it's the fallback if we ever have to turn off Experian

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

- Fixes #2388 
- Adds new logic to the duplicate org check, so that organizations with the same name but different states can still register.

## Changes Proposed

- Remove unused endpoints from AccountRequestController and its test
- Add logic to allow orgs with the same name but different states to register (the "duplicate" org will be registered with the state appended to its org name)
- Changes AccountRequestControllerTest to extend BaseFullStack instead of mocking individual services. This was required as I wanted to accurately test duplicate orgs without mocking the responses
- Add new tests to AccountRequestControllerTest to check for duplicate orgs and duplicate users
- Add a new Error class and logic in DemoOktaRepository to catch duplicate users (which don't typically occur within the same org but rather across our entire Okta group)
- Updates BaseFullStackTest to grant access to OrganizationService and DemoOktaRepository, which are used for assertions in AccountRequestControllerTest

## Additional Information

Here's a sample workflow for duplicate orgs in different states:
- `Central Schools` registers successfully in AZ. They're created with the org name `Central Schools`.
- A different `Central Schools` tries to register in WA. They're created with the org name `Central Schools-WA`.

Adding the state to the org name in the case of duplicate orgs isn't great, because it'll show up for users in the app, but it was the only solution that will still allow orgs to register. The database has a rule that organization names must be unique, so contacting support wouldn't really help the user in this case.

### UI
- n/a Any changes to the UI/UX are approved by design 
- n/a Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- n/a Database changes are submitted as a separate PR
  - n/a Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - n/a Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - n/a Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- n/a GraphQL schema changes are backward compatible with older version of the front-end

### Security
- n/a Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- n/a Any dependencies introduced have been vetted and discussed

## Cloud
- n/a DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 1]
```diff
@@ -211,114 +101,29 @@ public AccountResponse submitOrganizationAccountRequest(
...
     return _os.createOrganization(organizationName, organizationType, orgExternalId);
   }
 
-  private void checkForDuplicateOrg(String organizationName) {
+  private String checkForDuplicateOrg(String organizationName, String state) {
     Optional<Organization> potentialDuplicateOrg = _os.getOrganizationByName(organizationName);
     if (potentialDuplicateOrg.isPresent()) {
-      throw new BadRequestException("Organization is a duplicate.");
+      // potential duplicate orgs must be in the same state to be a true duplicate
+      if (potentialDuplicateOrg.get().getExternalId().startsWith(state)) {
```
Comment (timothybest-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:116]: FYI I think there are some `organizations` with out states in their external ID.

**Review by zdeveloper — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 2]
```diff
@@ -211,114 +101,29 @@ public AccountResponse submitOrganizationAccountRequest(
...
-        facilityAddress,
-        Translators.parsePhoneNumber(request.getFacilityPhoneNumber()),
-        null,
-        deviceSpecimenTypes,
-        providerName,
-        providerAddress,
-        Translators.parsePhoneNumber(request.getOpPhoneNumber()),
-        request.getNpi());
-  }
-
```
Comment (bmaden-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:301]: Woot woot!  Nice to see this big blocks of red lines.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 3]
```diff
@@ -330,36 +135,10 @@ private void createAdminUser(String firstName, String lastName, String email, St
     _aus.createUser(email, adminName, externalId, Role.ADMIN);
   }
 
-  // This is for temporary compatibility so the request can contain either the type label (old way)
-  // or the type name (new way).  Once the request is updated, then we only need to validate the
-  // type with `parseOrganizationType`
-  private String getTypeFromLabelOrValue(String labelOrValue) {
-    try {
-      return Translators.parseOrganizationTypeFromName(labelOrValue);
```
Comment (bmaden-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:338]: I think we can also get rid of `parseOrganizationTypeFromName` now if we want to.

**Review by bmaden-usds — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java (hunk 1 above)
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:117]: Yeah, I was worried that might be the case with older orgs. We don't really have a way to check the org's state in that case, so I just added a catch for the `UnexpectedRollbackException` that re-throws it as a BadRequest. It won't allow duplicate org names across states, but at least users will get a toast explaining what happened.

File: backend/src/test/java/gov/cdc/usds/simplereport/api/AccountRequestControllerTest.java [hunk 4]
```diff
@@ -167,166 +109,37 @@ void waitlistValidatesInput() throws Exception {
...
-            "707 White Milton Extension",
-            "Apt 3",
-            "Reprehenderit nostr",
-            "RI",
-            "13046",
-            "facility");
+    // then
+    Optional<Organization> org = _orgService.getOrganizationByName("Day Hayes Trading");
+    assertThat(org).isPresent();
+    assertThat(org.get().getExternalId().startsWith("RI-Day-Hayes-Trading-"));
```
[the remark this answers is not in the document]
Comment (emmastephenson) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/AccountRequestControllerTest.java]: Thanks, I hadn't seen the sonar notice! Fixed both instances

**Review by nickclyde — APPROVED** [review]
Looks good!!
